modules/@apostrophecms/piece-type/index.js

```diff
@@ -268,7 +268,17 @@ module.exports = {
             // Not subject to draft/publish workflow
             throw self.apos.error('invalid');
           }
-          return self.revertPublishedToPrevious(req, published);
+          const result = await self.revertPublishedToPrevious(req, published);
+          if (result) {
+            return {
```

Comment:
I guess the response also includes a status code? This seems purpose built for internal API, but the `else` one in particular seems to be less so for project use. Someone would need to look for `!lastPublished` or something. Wasn't it previously returning the reverted doc?

Reply:
... Yeah.

Looking at this, I think I can go back to returning the reverted published doc, or `false` if there is no longer a published doc. And the front end can go back to being responsible for deciding what message to show. That still provides access to `lastPublishedAt`. I'll try that.